app/src/ui/app.tsx

```diff
@@ -2441,13 +2446,55 @@ export class App extends React.Component<IAppProps, IAppState> {
         description={__DARWIN__ ? 'Current Repository' : 'Current repository'}
         tooltip={tooltip}
         foldoutStyle={foldoutStyle}
+        onContextMenu={this.onContextMenu}
         onDropdownStateChanged={this.onRepositoryDropdownStateChanged}
         dropdownContentRenderer={this.renderRepositoryList}
         dropdownState={currentState}
       />
     )
   }
 
+  private onContextMenu = () => {
+    const selection = this.state.selectedState
+
+    const repository = selection ? selection.repository : null
+    if (!repository) {
+      return
+    }
```

Comment:
We try to avoid relying on truthy/falsy checks, could we make this an explicit check instead and perhaps leverage optional chaining while we're at it?
```suggestion
    const repository = this.state.selectedState?.repository
    if (repository === undefined) {
      return
    }
```